quickevent/app/plugins/Competitors/src/competitorwidget.cpp

```diff
@@ -413,4 +427,28 @@ QVector<int> CompetitorWidget::veteranAges()
 	return ret;
 }
 
+bool CompetitorWidget::isCardInLentTable(int si_card)
+{
+	qfs::Query q;
+	q.exec("SELECT siId FROM lentcards WHERE siId=" + QString::number(si_card), qf::core::Exception::Throw);
+	return q.next();
+}
+
+// check if lentcard checkbox is checked and update lentcards table accordingly
+void CompetitorWidget::updateLentcardsTable()
+{
+	int si_card = ui->edSiId->siid();
+	bool isLent = ui->chkLentCard->isChecked();
+	bool inLentTable = isCardInLentTable(si_card);
+	qfs::Query q;
+	if (isLent && !inLentTable)
+	{
+		q.exec("INSERT INTO lentcards (siId) VALUES ('" + QString::number(si_card) + "')", qf::core::Exception::Throw);
+	}
+	if (!isLent && inLentTable)
+	{
+		q.exec("DELETE FROM lentcards WHERE siId= " + QString::number(si_card), qf::core::Exception::Throw);
```

Comment:
je to cely trochu spatne, my treba mame sadu cipu, ktery pujcujeme, jejich cisla si vyplnim do tabulky a pak uz nic nezadavam, protoze kdyz nekdo pribehne s cipem z tabulkym je pujceny